cupy/testing/_parameterized.py

```diff
@@ -1,19 +1,22 @@
 import itertools
+import re
 import types
-import typing as tp  # NOQA
 import unittest
 
 from cupy.testing import _bundle
 from cupy.testing import _pytest_impl
 
 
+_bad_name = re.compile(r' at 0x[0-9a-f]*')
+
+
 def _param_to_str(obj):
     if isinstance(obj, type):
         return obj.__name__
     elif hasattr(obj, '__name__') and isinstance(obj.__name__, str):
         # print __name__ attribute for classes, functions and modules
         return obj.__name__
-    return repr(obj)
+    return _bad_name.sub('', repr(obj))
```

Comment:
The other source of randomness is `repr` of param.  Examples:
```diff
-tests/cupyx_tests/fallback_mode_tests/test_fallback.py::TestDocs_param_3_{object=<numpy = <function eig at 0x7f197d6d8ca0>, cupy = None>}::test_docs
+tests/cupyx_tests/fallback_mode_tests/test_fallback.py::TestDocs_param_3_{object=<numpy = <function eig at 0x7f7d606f3ca0>, cupy = None>}::test_docs

-tests/cupyx_tests/fallback_mode_tests/test_notifications.py::TestNotificationModes_param_0_{func=<numpy = <function array_equiv at 0x7f197de3c700>, cupy = None>, shape=(3, 4)}::test_notification_ignore
+tests/cupyx_tests/fallback_mode_tests/test_notifications.py::TestNotificationModes_param_0_{func=<numpy = <function array_equiv at 0x7f7d680b0700>, cupy = None>, shape=(3, 4)}::test_notification_ignore

-tests/cupyx_tests/scipy_tests/ndimage_tests/test_filters.py::TestGenericFilter::test_filter[_param_0_{dtype=uint8, filter='generic_filter', footprint=False, func_or_kernel=(<cupy._core.raw.RawKernel object at 0x7f188c83ff40>, <function rms_pyfunc at 0x7f188c7d65e0>), ksize=3, mode='reflect', origin=0, output=None, shape=(4, 5)}]
+tests/cupyx_tests/scipy_tests/ndimage_tests/test_filters.py::TestGenericFilter::test_filter[_param_0_{dtype=uint8, filter='generic_filter', footprint=False, func_or_kernel=(<cupy._core.raw.RawKernel object at 0x7f7c6f881f40>, <function rms_pyfunc at 0x7f7c6f8175e0>), ksize=3, mode='reflect', origin=0, output=None, shape=(4, 5)}]
```

Maybe the 3 test files should be fixed instead.

Reply:
I guess this may generate duplicate test names when lambda is parameterized (e.g., `<function <lambda> at 0x7fba03090200>`) so I think fixing tests is better. Let me take care of it.

Reply:
#5727 #5728